lib/plugins/aws/package/compile/events/stream/index.js

```diff
@@ -134,6 +157,49 @@ class AwsCompileStreamEvents {
               if (typeof event.stream.enabled !== 'undefined') {
                 Enabled = event.stream.enabled ? 'True' : 'False';
               }
+              if (typeof event.stream.consumer !== 'undefined') {
+                if (typeof event.stream.consumer === 'boolean') {
+                  if (event.stream.consumer !== true) {
+                    throw new this.serverless.classes.Error(
+                      this.resolveInvalidKinesisConsumerErrorMessage(functionName)
+                    );
+                  }
+                } else if (typeof event.stream.consumer === 'string') {
+                  if (event.stream.consumer.indexOf('arn:') < 0) {
+                    throw new this.serverless.classes.Error(
+                      this.resolveInvalidKinesisConsumerErrorMessage(functionName)
+                    );
+                  }
+                } else if (typeof event.stream.consumer === 'object') {
+                  if (
+                    Object.keys(event.stream.consumer).length !== 1 ||
+                    !(
+                      _.has(event.stream.consumer, 'Fn::ImportValue') ||
+                      _.has(event.stream.consumer, 'Fn::GetAtt') ||
+                      (_.has(event.stream.consumer, 'Ref') &&
+                        (_.has(
+                          this.serverless.service.resources.Parameters,
+                          event.stream.consumer.Ref
+                        ) ||
+                          _.has(
+                            this.serverless.service.resources.Resources,
+                            event.stream.consumer.Ref
+                          ))) ||
+                      _.has(event.stream.consumer, 'Fn::Join')
+                    )
+                  ) {
+                    const errorMessage = [
+                      `'${JSON.stringify(
+                        event.stream
+                      )}'Bad dynamic property on stream event in function "${functionName}"`,
+                      ' If you use a dynamic "consumer" (such as with Fn::GetAtt, Fn::Join, Ref',
+                      ' or Fn::ImportValue) there must only be one key (either Fn::GetAtt, Fn::Join, Ref',
+                      ' or Fn::ImportValue) in the consumer object. Please check the docs for more info.',
+                    ].join('');
```

Comment:
I believe this validation is already made on AWS side. As we just pass through this value (and not try to interpret it for our needs) I would leave it without local validation here.